imports/node-app/plugins/discount-codes/mutations/applyDiscountCodeToCart.js

```diff
@@ -0,0 +1,134 @@
+import SimpleSchema from "simpl-schema";
+import Random from "@reactioncommerce/random";
+import ReactionError from "@reactioncommerce/reaction-error";
+import getCart from "../util/getCart.js";
+
+const inputSchema = new SimpleSchema({
+  cartId: String,
+  discountCode: String,
+  shopId: String,
+  token: {
+    type: String,
+    optional: true
+  }
+});
+
+/**
+ * @method applyDiscountCodeToCart
+ * @summary Applies a discount code to a cart
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} input - an object of all mutation arguments that were sent by the client
+ * @param {Object} input.cartId - Cart to add discount to
+ * @param {Object} input.discountCode - Discount code to add to cart
+ * @param {String} input.shopId - Shop cart belongs to
+ * @param {String} [input.token] - Cart token, if anonymous
+ * @returns {Promise<Object>} An object with the updated cart with the applied discount
+ */
+export default async function applyDiscountCodeToCart(context, input) {
+  inputSchema.validate(input);
+
+  const { cartId, discountCode, shopId, token } = input;
+  const { collections, userHasPermission, userId } = context;
+  const { Cart, Discounts } = collections;
+
+  // TODO: figure out the correct permission check here
+  // Should it be `discounts`, or `cart`?
+  // How do we determine this check if the user is the cart owner?
+  if (!userHasPermission(["admin", "owner", "discounts"], shopId)) {
+    throw new ReactionError("access-denied", "Access Denied");
+  }
```

Comment:
This one seems unnecessary given that `getCart` will already limit to carts that belong to the account/token, and then you have a separate admin check below it.